photon-api/src/main/scala/com/linkedin/photon/ml/function/glm/SingleNodeGLMLossFunction.scala

```diff
@@ -144,26 +146,43 @@ object SingleNodeGLMLossFunction {
    *
    * @param configuration The optimization problem configuration
    * @param singleLossFunction The PointwiseLossFunction providing functionality for l(z, y)
+   * @param priorModelOpt Optional prior model, required if this is an objective function for incremental training
    * @param interceptIndexOpt The index of the intercept, if there is one
+   * @param isIncrementalTrainingEnabled Is this an objective function for incremental training?
    * @return A new SingleNodeGLMLossFunction
    */
   def apply(
       configuration: GLMOptimizationConfiguration,
       singleLossFunction: PointwiseLossFunction,
-      interceptIndexOpt: Option[Int] = None): SingleNodeGLMLossFunction = {
+      priorModelOpt: Option[GeneralizedLinearModel] = None,
+      interceptIndexOpt: Option[Int] = None,
+      isIncrementalTrainingEnabled: Boolean = false): SingleNodeGLMLossFunction = {
```

Comment:
Why do we need both the `priorModelOpt` and the `isIncrementalTrainingEnabled` flag? In the match statement below, the only two valid cases are `(Some, true)` and `None, false`, meaning that if the `priorModelOpt` is present at all it implies that we're doing incremental training.